scio-bigquery/src/main/scala/com/spotify/scio/bigquery/syntax/ScioContextSyntax.scala

```diff
@@ -146,6 +147,9 @@ final class ScioContextOps(private val self: ScioContext) extends AnyVal {
     }
   }
 
+  def typedBigQueryTable[T: Schema: Coder](table: Table): SCollection[T] =
```

Comment:
We could also provide a:

```scala
def typedBigQueryTable[T: Schema: Coder](table: Table, parseFn: SchemaAndRecord => T): SCollection[T]
```

similar to what we provide in the avro api and also add Storage API support